packtools/sps/validation/dates.py

```diff
@@ -0,0 +1,121 @@
+from datetime import date
+import logging
+
+from packtools.sps.models.dates import ArticleDates
+
+
+def date_dict_to_date(date_dict):
+    return date(int(date_dict['year']), int(date_dict['month']), int(date_dict['day']))
+
+
+class ArticleDatesValidator:
+    def __init__(self, xmltree):
+        self.history = ArticleDates(xmltree)
+
+    def history_dates_are_complete(self):
+        response = []
+        for history_date in self.history.history_dates_list:
+            date_validated = is_complete(history_date, history_date['type'])
+            response.append(date_validated)
+
+        return response
+
+    def history_dates_are_sorted(self, order, required_events):
+        """
+        Checks the chronological order of occurrence dates of document publishing events.
+
+        Parameters
+        ----------
+        order : list
+            A list with the order in which events occur.
+        required_events : list
+            A list with required events.
+
+        Returns
+        -------
+        dict
+            A dictionary that registers the input data, eventual messages, the result and the expected and found orders.
+
+        Examples
+        --------
+        >>> dates_are_sorted(["received", "rev-request", "rev-recd", "accepted", "approved",],
+        ["received", "approved"])
+
+        {
+            'input': {
+                'order_of_events': ["received", "rev-request", "rev-recd", "accepted", "approved"],
+                'required_events': ["received", "approved"],
+                'events_date': [
+                {'rev-request': date(1998, 3, 14)},
+                {'rev-recd': date(1998, 5, 24)},
+                {'accepted': date(1998, 6, 6)},
+                {'approved': date(2012, 6, 1)}
+                ]
+            },
+            'message': ['the event received is required'],
+            'result': 'error',
+            'expected_order': [date(1998, 3, 14), date(1998, 5, 24), date(1998, 6, 6), date(2012, 6, 1)],
+            'found_order': [date(1998, 3, 14), date(1998, 5, 24), date(1998, 6, 6), date(2012, 6, 1)]
+        }
+        """
+        seq = []
+        history_dates = self.history.history_dates_dict
+        result = {
+            'input': {
+                'order_of_events': order,
+                'required_events': required_events,
+                'events_date': [{key: date_dict_to_date(value)} for key, value in history_dates.items()]
```

Comment:
@Rossi-Luciano No lugar de `[{key: date_dict_to_date(value)} for key, value in history_dates.items()]`, use o conteúdo de `self.history.history_dates_dict` ou `self.history.history_dates_list`, ou seja:

```python
                'events_date': history_dates
```